src/loot-boxes/loot-boxes.service.ts

```diff
@@ -60,59 +64,47 @@ export class LootBoxesService {
     );
   }
 
-  async getLocationBy(id: string): Promise<Location> {
-    const lootBoxWithLocation = await this.prisma.lootBox.findUnique({
-      where: {
-        id,
-      },
-      include: {
-        location: true,
-      },
-    });
-    if (!lootBoxWithLocation) {
-      throw new LootBoxesError(
-        LootBoxesError.NOT_FOUND,
-        'The loot box can not be found.',
-      );
-    }
-    if (!lootBoxWithLocation.location) {
-      return null;
-    }
-    return Location.create(lootBoxWithLocation.location);
-  }
+  async claimLootBox(
+    email: string,
+    address: string,
+    lootBoxId: string,
+  ): Promise<LootBox> {
+    const lootBox = await this.findOneById(lootBoxId);
 
-  async claimLootBox(email: string, lootBoxId: string): Promise<LootBox> {
-    const lootBox = await this.prisma.lootBox.findUnique({
-      where: { id: lootBoxId },
-    });
-    if (!lootBox) {
+    if (!lootBox.lootId) {
       throw new LootBoxesError(
-        LootBoxesError.NOT_FOUND,
-        'The loot box can not be found',
+        LootBoxesError.NO_LOOT_TO_CLAIM,
+        'This loot box does not contain any loot to claim',
       );
     }
-    if (lootBox.openedById !== null) {
+    if (lootBox.lootClaimed) {
       throw new LootBoxesError(
         LootBoxesError.ALREADY_CLAIMED,
-        'This loot box has already been claimed.',
-      );
-    }
-    const hasBeenFound = await this.hasBeenScanned(lootBoxId);
-    if (!hasBeenFound) {
-      throw new LootBoxesError(
-        LootBoxesError.FORBIDDEN,
-        'This lootbox has not been scanned yet and cannot be claimed.',
+        'This loot box has already been claimed',
       );
     }
+
+    const loot = await this.lootsService.findOneById(lootBox.lootId);
+    const eventBrand = await this.eventsService.getEventBrand(lootBox.eventId);
+    const brandRepoSlug = formatStringToSlug(eventBrand);
+    const tokenURI = this.configService
+      .get<string>('DOS_CDN')
+      .concat('/', brandRepoSlug, '/', loot.name, '.json');
+    await this.lootsService.decreaseCirculatingSupplyByOne(lootBox.lootId);
     const claimedLootBox = await this.setAndCreateWinner(lootBoxId, email);
-    const loot = await this.lootsService.getOneById(lootBox.lootId);
-    const lootImgUrl =
-      this.configService.get<string>('DOS_CDN') + '/' + loot.name + '.png';
-    await this.emailService.sendWinnerConfirmation(
+    const nftHasBeenMinted = await this.web3Service.mintNFT(address, tokenURI);
+    const emailHasBeenSent = await this.emailService.sendWinnerConfirmation(
       email,
-      lootImgUrl,
+      loot.imageUrl,
       loot.displayName,
     );
+    if (!emailHasBeenSent) {
+      console.log('Mail not send');
+    }
+    if (!nftHasBeenMinted) {
+      console.log('NFT not minted');
+    }
+
```

Comment:
1. Mint
2. Update db on mint success
3. Send email

* Throw error when mint not successfull.